cheatsheets/Credential_Stuffing_Prevention_Cheat_Sheet.md

```diff
@@ -1,84 +1,114 @@
 # Introduction
 
-[Credential stuffing](https://www.owasp.org/index.php/Credential_stuffing) is the automated injection of breached username/password pairs (typically from other sites) in order to identify accounts on the target system that use the same credentials. 
+This cheatsheet covers defences against two common types of authentication-related attacks: credential stuffing and password spraying. Although these are separate, distinct attacks, in many cases the defences that would be implemented to protect against them are the same, and they would also be effective at protecting against brute-force attacks. A summary of these different attacks is listed below:
 
-This is a subset of the brute force attack category: large numbers of spilled credentials are automatically entered into websites looking for matches to existing accounts, which the attacker can then hijack for their own purposes.
+| Attack Type | Description |
+|-------------|-------------|
+| Brute Force | Testing multiple passwords from dictionary or other source against a single account. |
+| Credential Stuffing | Testing username/password pairs obtained from the breach of another site. |
+| Password Spraying | Testing a single weak password against a large number of different accounts.|
 
-# Primary Defenses
+# Multi-Factor Authentication
 
-It should be noted that defense mechanisms are intended to be used in a layered approach. In most cases, a single defense option would be inadequate to stop most Credential Stuffing attacks.
+Multi-factor authentication (MFA) is by far the best defense against the majority of password-related attacks, including credential stuffing and password spraying, with analysis by Microsoft suggesting that it would have stopped [99.9% of account compromises](https://techcommunity.microsoft.com/t5/Azure-Active-Directory-Identity/Your-Pa-word-doesn-t-matter/ba-p/731984). As such, it should be implemented wherever possible; however, depending on the audience of the application, it may not be practical or feasible to enforce the use of MFA.
 
-In many cases, brute force protections will overlap with credential stuffing defenses.
+In order to balance security and usability, multi-factor authentication can be combined with other techniques to require for 2nd factor only in specific circumstances where there is reason to suspect that the login attempt may not be legitimate, such as a login from:
 
-Keep in mind that application can have different security levels for different users/roles or actions. For example for casual customers multi-factor authentication may be optional but for admins or some other special roles it should be enforced.
+- A new browser/device or IP address.
+- An unusual country or location.
+- Specific countries that are considered untrusted.
+- An IP address that appears on known blacklists.
+- An IP address that has tried to login to multiple accounts.
+- A login attempt that appears to be scripted rather than manual.
```

Comment:
For specific business action, a second authentication factor might be needed like for example banking transaction on a new account, update a beneficiary...It can be interesting to add the point that business feature can lead to MFA...

Reply:
I like the use of MFA on that kind of sensitive functionality (and also stuff like changing passwords or recovery email addresses). I'm not sure it really fits in too well here, but certainly something to include in the new MFA cheatsheet (#96).